JCR-2415 :Allow to clean swap files after re-indexing data

aboughzela opened the pull request:
Swap file is used as temporary location where the binary value will be spooled if no value storage is configured and the  max-buffer-size is exceeded.
 * It's possible to force the clean swap files thanks to the System property exo.jcr.spoolConfig.force.clean.swap. The default value of this parameter is false (force clean disabled).
*  If force clean is enabled, It's possible to define the live time (in seconds) of swap files thanks to the System property exo.jcr.spoolConfig.swap.live.time.  The default value of this parameter is -1 (disabled).
* If force clean swap is disabled, swap files is removed until cached binary property data is evicted or reload data container.

Commit:
Update POMs to version 1.17.x-SNAPSHOT

aboughzela committed:
POM update 1.17.x-SNAPSHOT

Commit:
JCR-2432 : fix all javadoc errors raised by doclint (JDK 8)

Commit:
PLF-6552: Upgrade cf-parent 14 -> 15-SNAPSHOT

aboughzela committed:
JCR-2433 : Ability to disable the BloomFilter

aboughzela committed:
JCR-2433  :  remove unit test.

aboughzela committed:
JCR-2435 : Wrong location of the search index.

aboughzela committed:
JCR-2436 : Improve re-indexing for Aggregated JCR Node.

aboughzela committed:
JCR-2415 :Allow to clean swap files after re-indexing data

Commit:
Ignore pom.xml.versionsBackup files

aboughzela committed:
JCR-2438 : Cannot delete node with inconsistent value data.

aboughzela committed:
JCR-2428 : Parallel node creation corrupts cache result count

wkhessairi committed:
[JCR-2437]Implementing hashcode() method

aboughzela committed:
JCR-2437 : Improve Implementing hashcode() method

aboughzela committed:
JCR-2442 : Remove Mixin from node contain Residual property

aboughzela committed:
JCR-2444 : Move IgnoreAccentAnalyzer from ECMS to JCR.

aboughzela committed:
JCR-2415 : Improve Swap File release

aboughzela committed:
JCR-2446 : Reduce loading time of NodeTypeDefinition

Commit:
SWF-3650: Update cf-parent to 15-M01

Commit:
SWF-3650: [eXoR] Update SNAPSHOT dependencies to RELEASE dependencies before Release.

Commit:
[maven-release-plugin] [SWF-3650] prepare release 1.17.0-M01

Commit:
[maven-release-plugin] [SWF-3650] prepare for next development iteration

Commit:
SWF-3650: [eXoR] Update RELEASE dependencies to SNAPSHOT dependencies after Release.

aboughzela committed:
JCR-2450 : Speed up cache initialization

exo-swf committed:
SWF-3658: [exo-release] Update SNAPSHOT dependencies to RELEASE dependencies before Release.

exo-swf committed:
[exo-release](mgreau) SWF-3658: prepare release 1.17.0-M02

exo-swf committed:
[exo-release](mgreau) SWF-3658: prepare for next development iteration

exo-swf committed:
SWF-3658: [exo-release] Update RELEASE dependencies to SNAPSHOT dependencies after Release.

Commit:
PLF-6630 : remove unused module containing fckeditor

bourasom committed:
Squashed commit of the following:

commit ce87bce59d0a90e9c9b0d9a6856598a72302a21f
Author: bourasom <obouras@exoplatform.com>
Date:   Fri Jun 24 10:57:04 2016 +0100

    ECMS-7453: add unit test

commit 8db76d713537e2b3da4b9a2a4cd5fbe0b02b1dfb
Author: bourasom <obouras@exoplatform.com>
Date:   Thu Jun 23 15:51:04 2016 +0100

    ECMS-7453:XXE issue with xml data importing function of ecms

thomasdelhomenie commented on exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java:
```diff
@@ -1054,6 +1059,50 @@ private void reindex(boolean doReindexing, boolean doForceReindexing, boolean do
+                     }
+                     catch (IOException e)
+                     {
+                        if(log.isDebugEnabled())
```
This condition should be removed. The log must be displayed whenever something goes wrong and it should be an error, not a warn IMO.

aboughzela replied:
change to error level, the error occur if the file doesn't existing any more when we try to remove it
(removed by another Thread before), So the log it's displayed only for debug level

thomasdelhomenie commented on exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java:
```diff
@@ -1054,6 +1059,50 @@ private void reindex(boolean doReindexing, boolean doForceReindexing, boolean do
+                     {
+                        if(log.isDebugEnabled())
+                        {
+                           log.warn("unable to delete swap files after re-index " + WorkspaceName);
```
Do not swallow the exception

aboughzela replied:
it's ok to change to error level, the error occur if the file doesn't existing any more when we try to remove it
(removed by another Thread before), So the log it's displayed only for debug level

thomasdelhomenie commented on exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/dataflow/SpoolConfig.java:
```diff
@@ -37,8 +38,43 @@
+         }
+         catch (NumberFormatException nex)
+         {
+            liveTime = -1;
```
We should add a log here (warn) to let the customer know that its setting is not correct.

aboughzela replied:
ok

thomasdelhomenie commented on exo.jcr.docs/exo.jcr.docs.developer/en/src/main/docbook/en-US/modules/jcr/configuration/workspace-persistence-storage.xml:
```diff
@@ -134,6 +134,15 @@
+     It's possible to force the clean swap files thanks to the System property exo.jcr.spoolConfig.force.clean.swap.
+     The default value of this parameter is false (force clean disabled).
+     If force clean is enabled, It's possible to define the live time (in seconds) of swap files thanks to the System property exo.jcr.spoolConfig.swap.live.time.
+     The default value of this parameter is -1 (disabled).
```
We should add :
- what happens if we do not enabled this property
- what is the role of these swap files
- what are the benefits/drawbacks of setting a low/high value for exo.jcr.spoolConfig.swap.live.time

If will help the customer to decide which values to set.

aboughzela replied:
Description updated

Commit:
SWF-3657: Prefix POMs name by eXo PLF::

Commit:
RELEASE-109: use cf-parent:15-RC01 to deploy in eXo Nexus

Commit:
JCR-2453 : make JCR quota test more determinist by using quotas relative to the data current size

Commit:
PAR-365: Update to cf-parent:15-RC02 for Java 8

Commit:
SWF-3837: Update version from 1.17.x to 1.18.x

aboughzela committed:
JCR-2415 : Improve Swap File release [apply given Remarks]

Commit:
SWF-3869: Update all PLF components versions to 5.0.x-SNAPSHOT version

bourasom committed:
JCR-2416:Strange behavior of indexing rule (#26)

JCR-2416:Strange behavior of indexing rule & fix Incorrect excerpt for index aggregates

aboughzela committed:
Fix/jcr 2454 (#25)

JCR-2454 : Error JCR occured after delete a base version

Commit:
SWF-3869: Update cf-parent to 16-SNAPSHOT

aboughzela committed:
PLF-7169 : Upgrade eXo Platform to ISPN last version (8)

fix exo-services version

Commit:
RELEASE-258: Update cf-parent to 16-M01

Commit:
SWF-3886: Add LGPL license file

thomasdelhomenie committed:
JCR-2456 : merge project jcr-services into project jcr (#32)

Commit:
JCR-2456: Fix artifact version after merge of jcr-services project

boubaker committed:
PLF-7233 Upgrade groovy to latest version (#30)

aboughzela committed:
Merge branch 'feature/infinispan-8' into develop

Commit:
PLF-7169: Fix pom version

thomasdelhomenie committed:
PLF-7349 : change log level to debug in GroovyScript2RestLoader when a groovy script is already loaded, since it is a normal case (#33)

aboughzela committed:
JCR-2459 : Failed recovery index on local index (#35)

aboughzela committed:
JCR-2458  : [index recovery strategy] rsync strategy implementation

Commit:
PLF-7397 : do not use URL to work with files in tests, use inputstreams instead

nttuyen committed:
Merge pull request #36 from exodev/fix/PLF-7397

PLF-7397 : do not use URL to work with files in tests, use inputstreams instead

Commit:
PLF-7414 : check indexing rules to decide wether to index or not binary fields

thomasdelhomenie committed:
PLF-7554 : upgrade to hsqldb 2.4.0 (#38)

aboughzela committed:
JCR-2462 : Keep old index up to date while a reindexation (#39)

Commit:
PLF-7414 : add a reindex method in SearchManager to return a Future

Commit:
PLF-7414 : add logs

Commit:
PLF-7414 : shitdown executor service

Commit:
PLF-7414 : throw exception when a query fails

boubaker committed:
PLF-7411 fix javadoc

Commit:
Merge branch 'feature/files-in-es' into develop

Commit:
RELEASE-510: upgrade parent pom to 16-RC01

Commit:
SWF-4200: Update projects versions to 5.1.x-SNAPSHOT for next development

boubaker committed:
JCR-2464 fix async reindex when removing node (#42)

boubaker committed:
JCR-2463 File name too long error on values while reindexing (#44)

aboughzela committed:
ECMS-7692 : Permissions changes on files are not reindexed (#43)

aboughzela committed:
ECMS-7602 : read-only opening file when no permission  doesn't work (#45)

aboughzela committed:
PLF-7889 : non-thread safe date formatting (#48)

Commit:
RELEASE-610: upgrade cf-parent dependency to 17-M01

aboughzela committed:
PLF-7902 : Restrict WebDAV folder listing only for some paths (#47)

aboughzela committed:
Merge branch 'develop' into fix/1.16.7-GA/JCR-2415

aboughzela committed:
JCR-2415 : simplify system property configuration

aboughzela committed:
JCR-2415 : simplify system property configuration

boubaker committed:
PLF-7866 Enhance NodeHierarchyCreator Service (#50)

aboughzela committed:
Merge branch 'develop' into fix/1.16.7-GA/JCR-2415

boubaker committed:
PLF-8034 Fix NodeHierarchyCreator startup (#51)

Commit:
SWF-4382: Upgrade cf-parent 17-M01 -> 17-RC01

Commit:
Merge branch 'develop' into fix/1.16.7-GA/JCR-2415
